zerver/tornado/event_queue.py

```diff
@@ -987,7 +986,9 @@ def get_client_payload(apply_markdown: bool, client_gravatar: bool) -> Dict[str,
             all_bot_user_ids=all_bot_user_ids,
         )
 
-        internal_data = asdict(user_notifications_data)
+        # Calling asdict is slow; pull the attributes out directly
+        internal_data = {**user_notifications_data.__dict__}
```

Comment:
More specifically, `asdict` does a deep copy ([documentation](https://docs.python.org/3/library/dataclasses.html#dataclasses.asdict)), which is presumably unnecessary.

`{**vars(user_notifications_data)}` is an alternative that avoids a direct magic method call.

Reply:
Thanks for the `vals` suggestion!  I updated the comment to mention the deep copy (it was already in the commit message) and to also switch to `vals`.